`mypy` raises errors after `numpy` 2.5.0

https://github.com/pandas-dev/pandas-stubs/actions/runs/27948511803/job/82699244836#step:4:18
```text
tests/scalars/test_scalars.py:571: error: Unused "type: ignore" comment  [unused-ignore]
tests/scalars/test_scalars.py:619: error: Unused "type: ignore" comment  [unused-ignore]
tests/scalars/test_scalars.py:907: error: Expression is of type "Any", not "ndarray[tuple[Any, ...], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:908: error: Expression is of type "Any", not "ndarray[tuple[Any, ...], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:910: error: Expression is of type "Any", not "ndarray[tuple[int, int], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:911: error: Expression is of type "Any", not "ndarray[tuple[int, int], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:913: error: Expression is of type "Any", not "ndarray[tuple[int], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:914: error: Expression is of type "Any", not "ndarray[tuple[int], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:1317: error: Expression is of type "Any", not "ndarray[tuple[Any, ...], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:1318: error: Expression is of type "Any", not "ndarray[tuple[Any, ...], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:1320: error: Expression is of type "Any", not "ndarray[tuple[int, int], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:1321: error: Expression is of type "Any", not "ndarray[tuple[int, int], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:1323: error: Expression is of type "Any", not "ndarray[tuple[int], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/scalars/test_scalars.py:1324: error: Expression is of type "Any", not "ndarray[tuple[int], dtype[numpy.bool[builtins.bool]]]"  [assert-type]
tests/arrays/test_datetime_array.py:81: error: Expression is of type "Any", not "DatetimeArray"  [assert-type]
tests/arrays/test_datetime_array.py:92: error: Expression is of type "Any", not "DatetimeArray"  [assert-type]
tests/arrays/test_datetime_array.py:107: error: Expression is of type "Any", not "DatetimeArray"  [assert-type]
tests/arrays/test_datetime_array.py:117: error: Expression is of type "Any", not "DatetimeArray"  [assert-type]
tests/arrays/test_datetime_array.py:123: error: Expression is of type "Any", not "DatetimeArray"  [assert-type]
tests/arrays/test_datetime_array.py:124: error: Expression is of type "Any", not "DatetimeArray"  [assert-type]
tests/arrays/test_datetime_array.py:313: error: Expression is of type "Any", not "DatetimeArray"  [assert-type]
tests/series/test_series.py:2644: error: Expression is of type "ndarray[tuple[int], dtype[datetime64[int]]]", not "ndarray[tuple[int], dtype[datetime64[Any]]]"  [assert-type]
tests/series/test_series.py:2650: error: Expression is of type "ndarray[tuple[int], dtype[timedelta64[int]]]", not "ndarray[tuple[int], dtype[timedelta64[Any]]]"  [assert-type]
tests/series/test_series.py:2657: error: Expression is of type "ndarray[tuple[int], dtype[timedelta64[int]]]", not "ndarray[tuple[int], dtype[timedelta64[Any]]]"  [assert-type]
Found 24 errors in 3 files (checked 345 source files)
```

--- a/tests/arrays/test_integer.py
+++ b/tests/arrays/test_integer.py
@@ -46,12 +46,8 @@ def test_construction_sequence(
 
     if TYPE_CHECKING:
         assert_type(pd.array([-2, 3]), IntegerArray)
-        # TODO: pandas-dev/pandas-stubs#1786 investigate and report to pyrefly
-        assert_type(  # pyrefly: ignore[assert-type]
-            pd.array([1 << 32, np.int8(1) << 6]), IntegerArray
-        )
 
-        assert_type(pd.array([2, np.int8(3)]), IntegerArray)
+        assert_type(pd.array([1 << 33, 2, np.int8(3)]), IntegerArray)
 
         assert_type(pd.array([5, np.int16(0o10), None]), IntegerArray)
         assert_type(pd.array([0xD, np.int16(21), pd.NA]), IntegerArray)

--- a/tests/scalars/test_scalars.py
+++ b/tests/scalars/test_scalars.py
@@ -1755,10 +1755,9 @@ def test_period_add_subtract() -> None:
     check(assert_type(as_np_td + p, pd.Period), pd.Period)
     check(assert_type(p.__radd__(as_np_td), pd.Period), pd.Period)
 
-    # TODO: pandas-dev/pandas-stubs#1786 investigate and report to pyrefly
-    check(
-        assert_type(as_np_i64 + p, pd.Period), pd.Period  # pyrefly: ignore[assert-type]
-    )
+    # TODO: facebook/pyrefly#4422
+    # pyrefly: ignore[assert-type]
+    check(assert_type(as_np_i64 + p, pd.Period), pd.Period)
     check(assert_type(p.__radd__(as_np_i64), pd.Period), pd.Period)
 
     check(assert_type(as_int + p, pd.Period), pd.Period)

--- a/tests/scalars/test_scalars.py
+++ b/tests/scalars/test_scalars.py
@@ -497,7 +497,7 @@ def test_timedelta_properties_methods() -> None:
     check(assert_type(td.to_pytimedelta(), dt.timedelta), dt.timedelta)
     check(assert_type(td.to_timedelta64(), np.timedelta64), np.timedelta64)
     check(assert_type(td.total_seconds(), float), float)
-    # TODO: pandas-dev/pandas-stubs#1786 remove the conditional warning
+    # TODO: pandas-dev/pandas#66608 remove the conditional warning
     with pytest_warns_conditioned(
         DeprecationWarning,
         r"The 'generic' unit for NumPy timedelta is deprecated",

